iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/GenericUDFIcebergBucket.java

```diff
@@ -158,6 +159,16 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
         };
         break;
 
+      case DATE:
+        converter = new PrimitiveObjectInspectorConverter.DateConverter(argumentOI,
+          PrimitiveObjectInspectorFactory.writableDateObjectInspector);
+        Transform<Integer, Integer> dateTransform = Transforms.bucket(Types.DateType.get(), numBuckets);
```

Comment:
Nit:   `Transform<T, Integer> bucket(Type type, int numBuckets)` will be removed in the future.
https://github.com/apache/iceberg/blob/e7d21a948865dc182a718f779757cb5f181856cb/api/src/main/java/org/apache/iceberg/transforms/Transforms.java#L196-L208

Maybe we can use this instread:
`Function<Object, Integer> dateTransform = Transforms.bucket(numBuckets).bind(Types.DateType.get());`

Reply:
changed